projects/addon-preview/components/preview/preview.component.ts

```diff
@@ -0,0 +1,253 @@
+        this.zoom$,
+        this.rotation$,
+    ]).pipe(
+        map(([translate, zoom, rotation]) =>
+            this.sanitizer.bypassSecurityTrustStyle(
+                `translate(${translate}) scale(${zoom}) rotate(${rotation}deg)`,
+            ),
+        ),
+    );
+
+    @ViewChild('contentWrapper')
+    set contentWrapper({nativeElement}: ElementRef<HTMLElement>) {
+        this.initClickSubscription(nativeElement);
+    }
+
+    constructor(
+        @Inject(DomSanitizer) private readonly sanitizer: DomSanitizer,
+        @Inject(ElementRef) readonly elementRef: ElementRef<HTMLElement>,
+        @Inject(TuiDestroyService) readonly destroy$: Observable<void>,
+        @Inject(TUI_PREVIEW_TEXTS)
+        readonly texts$: Observable<LanguagePreview['previewTexts']>,
+    ) {}
+
+    rotate() {
+        this.rotation$.next(this.rotation$.value - ROTATION_ANGLE);
+    }
+
+    onPan(delta: [number, number]) {
+        this.coordinates$.next(
+            this.getGuardedCoordinates(
+                this.coordinates$.value[0] + delta[0],
+                this.coordinates$.value[1] + delta[1],
+            ),
+        );
+    }
+
+    onMutation(contentWrapper: HTMLElement) {
+        const {clientWidth, clientHeight} = contentWrapper;
+
+        this.refresh(clientWidth, clientHeight);
+    }
+
+    onZoom({clientX, clientY, delta}: TuiZoom) {
+        if (this.zoomable) {
+            this.processZoom(clientX, clientY, delta);
+        }
+    }
+
+    onResize(contentResizeEntries: ReadonlyArray<ResizeObserverEntry>) {
+        if (contentResizeEntries.length === 0) {
+            return;
+        }
+
+        const {width, height} = contentResizeEntries[0].contentRect;
+
+        this.refresh(width, height);
+    }
+
+    reset() {
+        this.zoom$.next(this.minZoom);
+        this.coordinates$.next(EMPTY_COORDINATES);
+    }
+
+    private get offsets(): {offsetX: number; offsetY: number} {
+        const offsetX = ((this.zoom$.value - this.minZoom) * this.width) / 2;
+        const offsetY = ((this.zoom$.value - this.minZoom) * this.height) / 2;
+
+        return {offsetX, offsetY};
+    }
+
+    private calculateMinZoom(
+        contentHeight: number,
+        contentWidth: number,
+        boxHeight: number,
+        boxWidth: number,
+    ): number {
+        const bigSize =
+            contentHeight > boxHeight * INITIAL_SCALE_COEF ||
+            contentWidth > boxWidth * INITIAL_SCALE_COEF;
+
+        return bigSize
+            ? round(
+                  Math.min(
+                      (this.elementRef.nativeElement.clientHeight * INITIAL_SCALE_COEF) /
+                          contentHeight,
+                      (this.elementRef.nativeElement.clientWidth * INITIAL_SCALE_COEF) /
+                          contentWidth,
+                  ),
+                  2,
+              )
+            : 1;
+    }
+
+    private initClickSubscription(element: HTMLElement) {
```

Comment:
I don't like that we do this. If there's no way to make it cleaner I say let's just drop this feature. Sergey Novoselov doesn't mind.